packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx

```diff
@@ -382,9 +384,70 @@ const ViewThreadPage = ({ identifier }: ViewThreadPageProps) => {
     action: 'comment',
   });
 
+  const getMetaDescription = (meta: string) => {
+    try {
+      const parsedMeta = JSON.parse(meta);
+      if (getTextFromDelta(parsedMeta)) {
+        return getTextFromDelta(parsedMeta) || meta;
+      } else {
+        return meta;
+      }
+    } catch (error) {
+      return;
+    }
+  };
+
+  const ogTitle =
+    thread?.title?.length > 60
+      ? `${thread?.title?.slice?.(0, 52)}...`
+      : thread?.title;
+  const ogDescription =
+    getMetaDescription(thread?.body || '')?.length > 155
+      ? `${getMetaDescription(thread?.body || '')?.slice?.(0, 152)}...`
+      : getMetaDescription(thread?.body || '');
+  const ogImageUrl = app?.chain?.meta?.iconUrl;
+
   return (
     // TODO: the editing experience can be improved (we can remove a stale code and make it smooth) - create a ticket
     <>
+      <Helmet defaultTitle="Common">
```

Comment:
This seems like it should be a component that we can reuse.  It's a lot of logic that is out of place vs the original component.  I think you just have to pass it some metadata.

Reply:
yeah i agree, wasn't sure about the original reason this was done. Will update

Reply:
done